src/snapshots.js

```diff
@@ -199,7 +199,9 @@ export async function* takeStorybookSnapshots(percy, callback, { baseUrl, flags
     while (snapshots.length) {
       try {
         // use a single page to capture story snapshots without reloading
-        yield* withPage(percy, previewUrl, async function*(page) {
+        // loading an existing story instead of just iframe.html as that triggers `storyMissing` event
```

Comment:
one more refactoring change not related to current PR,
we had made were having `await` on the promise 
https://github.com/percy/percy-storybook/blob/254b3266ea3e645c87eab7b3430564cf30a49526/src/start.js#L35


instead we'll use yield, this is to support cancellation of promise.

Reply:
I tried rewriting this using generators, but I am unable to do os.
I am unable to find any good examples, the existing solution that I checked out on stackoverflow involved creating an internal array, where resolved promises push values to, and the generator yields from there.
https://stackoverflow.com/questions/51045136/how-can-i-use-a-event-emitter-as-an-async-generator

Open to suggestion if there is any other way to refactor this.

Reply:
changing `await` to `yield` works?

Reply:
not a blocker for this release.

let me refactor it alongside some other PR.